libs/owasp-diff-runner/src/main/java/ddf/lib/OwaspDiffRunner.java

```diff
@@ -0,0 +1,184 @@
+
+        try {
+            currentBranchName = IOUtils.toString(runTime.exec(GIT_SHORT_BRANCH_NAME_COMMAND)
+                    .getInputStream())
+                    .replace(File.separator, "")
+                    .replace(System.getProperty("line.separator"), "");
+
+            changedFiles = IOUtils.toString(runTime.exec(
+                    GIT_DIFF_NAME_COMMAND + currentBranchName + "..master")
+                    .getInputStream());
+        } catch (IOException e) {
+            throw new OwaspDiffRunnerException(OwaspDiffRunnerException.UNABLE_TO_RETRIEVE_GIT_INFO,
+                    e);
+        }
+
+        System.out.println(
+                "Comparing commits of branch " + currentBranchName + " to master. Changed poms: ");
+
+        return Arrays.stream(changedFiles.split(System.getProperty("line.separator")))
+                .filter(path -> path.endsWith("pom.xml"))
+                .peek(System.out::println)
+                .map(path -> path.endsWith(File.separator + "pom.xml") ?
+                        path.replace(File.separator + "pom.xml", "") :
+                        path.replace("pom.xml",
+                                File.separator)) //Special case for the root pom, change path pom.xml -> /
+                .collect(Collectors.joining(","));
+    }
+
+    private static String getMavenHome() throws OwaspDiffRunnerException {
+        String mavenHome;
+        String mavenVersionInfo;
+
+        try {
+            mavenVersionInfo = IOUtils.toString(runTime.exec(MAVEN_VERSION_COMMAND)
+                    .getInputStream());
+
+            //parsing console response, confirmed to work with at least maven version 3.3.9
+            mavenHome = Arrays.stream(mavenVersionInfo.split(System.getProperty("line.separator")))
+                    .filter(str -> str.contains("Maven home:"))
+                    .findFirst()
+                    .get()
+                    .split("Maven home:")[1].trim();
+
+        } catch (Exception e) {
+            throw new OwaspDiffRunnerException(OwaspDiffRunnerException.UNABLE_TO_RETRIEVE_MAVEN_HOME,
+                    e);
+        }
+
+        return mavenHome;
+    }
+
+    private static String getLocalRepo() throws OwaspDiffRunnerException {
+        String mavenHelpInfo;
+        String localRepoProperty = System.getProperty("maven.repo.local");
+
+        if (StringUtils.isNotEmpty(localRepoProperty)) {
+            if (!new File(localRepoProperty).exists()) {
+                throw new OwaspDiffRunnerException(
+                        OwaspDiffRunnerException.UNABLE_TO_RETRIEVE_LOCAL_MAVEN_REPO
+                                + localRepoProperty);
+            }
+            return localRepoProperty;
+        }
+
+        try {
+            mavenHelpInfo = IOUtils.toString(runTime.exec(MAVEN_SETTINGS_COMMAND)
+                    .getInputStream());
+
+            //parsing console response, confirmed to work with at least maven version 3.3.9
```

Comment:
This was one of my concerns but I did a search for the last time this was altered and it has not in a very long time, especially this local repo section. I couldn't find a environment independent way of getting the local repo path as well. I was hoping for a maven property that I could use but none seem to exist.